nsxt/resource_nsxt_policy_site.go

```diff
@@ -0,0 +1,267 @@
+package nsxt
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-gm/global_infra"
+	gm_model "github.com/vmware/vsphere-automation-sdk-go/services/nsxt-gm/model"
+)
+
+var siteTypeValues = []string{
+	"ONPREM_LM",
+	"SDDC_LM",
+}
+
+func resourceNsxtPolicySite() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicySiteCreate,
+		Read:   resourceNsxtPolicySiteRead,
+		Update: resourceNsxtPolicySiteUpdate,
+		Delete: resourceNsxtPolicySiteDelete,
+		Importer: &schema.ResourceImporter{
+			State: nsxtPolicyPathResourceImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"nsx_id":       getNsxIDSchema(),
+			"path":         getPathSchema(),
+			"display_name": getDisplayNameSchema(),
+			"description":  getDescriptionSchema(),
+			"revision":     getRevisionSchema(),
+			"tag":          getTagsSchema(),
+			"fail_if_rtep_misconfigured": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Fail onboarding if RTEPs misconfigured",
+				Default:     true,
+			},
+			"fail_if_rtt_exceeded": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Fail onboarding if maximum RTT exceeded",
+				Default:     true,
+			},
+			"maximum_rtt": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Description:  "Maximum acceptable packet round trip time (RTT)",
+				Default:      250,
+				ValidateFunc: validation.IntBetween(0, 1000),
+			},
+			"site_connection_info": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Connection information",
+				MaxItems:    3,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"fqdn": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "Fully Qualified Domain Name of the Management Node",
+						},
+						"password": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Sensitive:   true,
+							Description: "Password",
+						},
+						"site_uuid": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "id of Site",
+						},
+						"thumbprint": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "Thumbprint of Enforcement Point",
+						},
+						"username": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "Username",
+						},
+					},
+				},
+			},
+			"site_type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Persistent Site Type",
+				ValidateFunc: validation.StringInSlice(siteTypeValues, false),
```

Comment:
No default here? What happens if this is unspecified? Should the value be Computed in this case?

Reply:
[Spec](https://vdc-download.vmware.com/vmwb-repository/dcr-public/8e2a4b1a-659f-4be6-a4f4-7ca42b04cb84/9a2c1372-7174-40f0-89d2-27615bfd5d08/api_includes/types_Site.html) doesn't specify a default here.
As UI in GM allows only on-prem addition, maybe we could discard this attribute entirely?
![Screenshot 2024-02-15 at 12 10 55](https://github.com/vmware/terraform-provider-nsxt/assets/9676139/3cdee735-c77f-4a23-b92c-4a6504d0ae76)

Reply:
I think this depends on env, and perhaps default depends on env as well. If NSX auto-assigns this value, we can define it as `Computed`.

Reply:
Turns out this this is required...
```
│ Error:  Failed to create Site f5bb6846-cf9d-4efd-9cd6-4d11de593fff: Field level validation errors: {value  of property site_type is not one of the allowed values [ONPREM_LM, SDDC_LM]} (code 255)
```